### ISSUE TITLE
Assembly reference 'mscorlib.dll' was not found or is invalid

### ISSUE THREAD

**Issue by prosconi** [fsharp/FSharp.Compiler.Service#863]
Starting from a fresh classic .NET 4.5 console application. Trying to perform compilation using documentation from: https://fsharp.github.io/FSharp.Compiler.Service/compiler.html

```
open System.IO
open Microsoft.FSharp.Compiler.SourceCodeServices

[<EntryPoint>]
let main _argv = 

    let checker = FSharpChecker.Create()
    let fn = Path.GetTempFileName()
    let fn2 = Path.ChangeExtension(fn, ".fsx")
    let fn3 = Path.ChangeExtension(fn, ".dll")

    File.WriteAllText(fn2, """
module M

type C() = 
    member x.P = 1

let x = 3 + 4
    """)
    let args =
        [|
            "fsc.exe"
            "-o"; fn3
            "-a"; fn2 
        |]
    
    let errors1, exitCode1 = 
        checker.Compile(args)
        |> Async.RunSynchronously

    printfn "%A" errors1 // Assembly reference 'mscorlib.dll' was not found or is invalid
    0
```

#### Repro steps
1. Visual Studio 2017
2. New F# Console Application .NET Framework 4.5
3. Install FSCS via NuGet 
```<package id="FSharp.Compiler.Service" version="22.0.3" targetFramework="net45" />```
4. Paste in above code and inspect the errors from the Compile call

[FSCSMscorlib.zip](https://github.com/fsharp/FSharp.Compiler.Service/files/1935133/FSCSMscorlib.zip)

#### Expected behavior
No errors are expected

#### Actual behavior
One errors is returned 
```Assembly reference 'mscorlib.dll' was not found or is invalid```

#### Known workarounds
N/A

#### Related information 
* Windows 10
* .NET 4.5

**Comment by dsyme** [comment]
THanks for reporting the bug.  Giving an explicit mscorlib and FSharp.Core reference should do the trick as a workaround?

**Comment by prosconi** [comment]
Something interesting... while using procmon.exe I could see a bunch of areas where it was looking for mscorlib, one of those was `C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2`

Anyways, the below does work:
```
    let args =
        [|
            "fsc.exe"
            "--noframework"
            "-r"; @"C:\Program Files (x86)\Reference Assemblies\Microsoft\FSharp\.NETFramework\v4.0\4.4.3.0\FSharp.Core.dll"
            "-r"; @"C:\Windows\Microsoft.NET\Framework64\v4.0.30319\mscorlib.dll"
            "-o"; fn3
            "-a"; fn2 
        |]
```

Are there any guidelines on finding mscorlib?

**Comment by dsyme** [comment]
@prosconi If on .NET Framework then you can use the one from `GetRuntimeDirectory()`

**Comment by jaccarmac** [comment]
I've tried the workarounds on this but get weird behavior after the fact on Linux. The resulting DLL conflicts with the project I try to integrate it with due to a `System.Runtime` mismatch. I don't get the error on Windows and in fact don't even need to add mscorlib manually.

**Comment by fangyi-zhou** [comment]
I've run into this on macos when I try to use follow the typed AST example using a `FSharpChecker.Create`. I've tried setting `MONO_PATH`, adding `mscorlib` to `fsproj` file and got no success. Any suggestions on how to fix it?

**Comment by ErikSchierboom** [comment]
> I've run into this on macos when I try to use follow the typed AST example using a FSharpChecker.Create. I've tried setting MONO_PATH, adding mscorlib to fsproj file and got no success. Any suggestions on how to fix it?

@fangyi-zhou I have the same issue on macOS. Did you manage to fix it?

**Comment by jmquigs** [comment]
I was following the tutorial https://fsharp.github.io/FSharp.Compiler.Service/symbols.html and got this error on Ubuntu 19.10 with DN core 3.0.1.  I am using the Nuget package for the compiler service.

I was able to workaround by adding several dlls manually to the options returned by `GetProjectOptionsFromScript`:

```fsharp
    let projOptions, _errors = 
        checker.GetProjectOptionsFromScript(file, SourceText.ofString input)
        |> Async.RunSynchronously

    let runtimeDir = System.Runtime.InteropServices.RuntimeEnvironment.GetRuntimeDirectory()
    printfn "Runtime Dir: %A" runtimeDir

    let fprojOptions = projOptions

    let addLibFromRTDirIfMissing (libName:string) (opts:string[]) = 
      let hasIt = opts |> Array.tryFind (fun opt -> opt.EndsWith(libName))
      if hasIt.IsSome then 
        opts
      else 
        let lib = Path.Combine(runtimeDir, libName)
        if not (File.Exists(lib)) then failwithf "can't find %A in runtime dir: %A" libName runtimeDir
        printfn "Adding %A to options" lib
        Array.concat [ opts; [| "-r:" + lib |] ]

    let oo = 
      fprojOptions.OtherOptions 
      |> addLibFromRTDirIfMissing "mscorlib.dll" 
      |> addLibFromRTDirIfMissing "netstandard.dll"     
      |> addLibFromRTDirIfMissing "System.Runtime.dll"     
      |> addLibFromRTDirIfMissing "System.Runtime.Numerics.dll"
      |> addLibFromRTDirIfMissing "System.Private.CoreLib.dll"
      |> addLibFromRTDirIfMissing "System.Collections.dll"
      |> addLibFromRTDirIfMissing "System.Net.Requests.dll"
      |> addLibFromRTDirIfMissing "System.Net.WebClient.dll"
    let fprojOptions = { fprojOptions with OtherOptions = oo }
    checker.ParseAndCheckProject (fprojOptions)
    |> Async.RunSynchronously
```
